src/api/wrappers/cf_wrapper.py

```
@@ -0,0 +1,48 @@
+import fire
+import requests
+from json.decoder import JSONDecodeError
+from src.api.dataclasses.cf_response_objects import CFGetModResponse, CFGetModData, CFDataResponse
+
+
+class CurseForgeWrapper:
+  '''Wrapper for the CurseForge API.'''
+
+  def __init__(self, api_key):
+    self.api_key = api_key
+
+  def get_mod(self, mod_id: int) -> CFDataResponse:
+    '''
+    Return CFDataResponse object containing mod metadata as a CFGetModResponse object.
+
+    Parameters:
+    mod_id (int): The Mod ID for the mod metadata to be fetched
+
+    Returns:
+    CFGetModData: Object containing CFGetModResponse object or None, statusCode of API request, 
+    and status containing status or error message.
+    '''
+    get_mod_url = f'https://api.curseforge.com/v1/mods/{mod_id}'
+    headers = {
+        'Accept': 'application/json',
+        'x-api-key': self.api_key
+    }
+
+    try:
+      response = requests.get(get_mod_url, headers=headers)
+    except requests.exceptions.HTTPError as err:
+      raise fire.core.FireError(
+        f'A problem occurred while querying the CurseForge API for mod {mod_id}: HTTPError: {err.response.status_code} {err.response.reason}')
+    except requests.exceptions.RequestException as err:
+      raise fire.core.FireError(
+        f'A problem occurred while querying the CurseForge API for mod {mod_id}: RequestException: {err.args[0]}')
```

Suppress the error, hide the implementation details from the user.

---

Idea: subclass FireError into an error type that we can raise like the roof

Give it a parameter to indicate if the error is unrecoverable or strange that is worth logging.
If so, log it to disk somewhere so we can ask users to give us the logfile for fuckery.

Hide the implementation details from the user. The user doesn't care what the RequestExceptionFuckery is.